cpp/include/Ice/LocalExceptions.h

```diff
@@ -0,0 +1,1633 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+#ifndef ICE_LOCAL_EXCEPTIONS_H
+#define ICE_LOCAL_EXCEPTIONS_H
+
+#include "Ice/BuiltinSequences.h"
+#include "Ice/Identity.h"
+#include "Ice/Version.h"
+#include "LocalException.h"
+
+#include <memory>
+#include <string>
+#include <string_view>
+#include <utility>
+
+#if defined(__clang__)
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wshadow-field-in-constructor"
+#elif defined(__GNUC__)
+#    pragma GCC diagnostic push
+#    pragma GCC diagnostic ignored "-Wshadow"
+#endif
+
+// This file contains most exceptions derived from Ice::LocalException.
+
+namespace Ice
+{
+    /**
+     * This exception is raised if a request failed. This exception, and all exceptions derived from
+     * {@link RequestFailedException}, are transmitted by the Ice protocol, even though they are declared
```

Comment:
With doxygen `{@link ...}` is not required. It automatically creates links for all words that match a type name. We can cleanup those in a separate PR.